plugins/distance_table.hpp

```diff
@@ -70,24 +70,34 @@ template <class DataFacadeT> class DistanceTablePlugin final : public BasePlugin
     int HandleRequest(const RouteParameters &route_parameters,
                       osrm::json::Object &json_result) override final
     {
-        if (!check_all_coordinates(route_parameters.coordinates))
+        const bool useSameTgtSrc = route_parameters.coordinates.size() ? true : false;
+        if ((useSameTgtSrc && route_parameters.destinations.size()) || (useSameTgtSrc && route_parameters.sources.size()))
+        {
+            return 400;
+        }
+
+        if ((useSameTgtSrc && !check_all_coordinates(route_parameters.coordinates)) ||
+            (!useSameTgtSrc && !check_all_coordinates(route_parameters.destinations, 2) && !check_all_coordinates(route_parameters.sources, 1)))
         {
             return 400;
         }
 
         const auto &input_bearings = route_parameters.bearings;
-        if (input_bearings.size() > 0 && route_parameters.coordinates.size() != input_bearings.size())
+        unsigned nb_coordinates = useSameTgtSrc ? route_parameters.coordinates.size() : (route_parameters.destinations.size() + route_parameters.sources.size());
+        if (input_bearings.size() > 0 && nb_coordinates != input_bearings.size())
         {
             json_result.values["status"] = "Number of bearings does not match number of coordinates .";
             return 400;
         }
 
         const bool checksum_OK = (route_parameters.check_sum == facade->GetCheckSum());
-        unsigned max_locations =
+        unsigned max_locations = useSameTgtSrc ?
+            std::min(static_cast<unsigned>(max_locations_distance_table),
+                     static_cast<unsigned>(route_parameters.coordinates.size())) :
             std::min(static_cast<unsigned>(max_locations_distance_table),
-                     static_cast<unsigned>(route_parameters.coordinates.size()));
+                     static_cast<unsigned>(route_parameters.destinations.size()));
 
-        PhantomNodeArray phantom_node_vector(max_locations);
+        PhantomNodeArray phantom_node_target_vector(max_locations);
         for (const auto i : osrm::irange(0u, max_locations))
```

Comment:
You can move this loop to a utility function and like this:
```c++
GetCoordinates(const std::vector<FixedPointCoordinate>& inputs, PhantomNodeArray& output)
{
....
}
```

The code should then look somethings like:
```c++
GetCoordinates(route_parameters.coordinates, phantom_node_vector);
std::insert(phantom_node_source_vector.end(), phantom_node_vector.begin(), phantom_node_vector.end());
std::insert(phantom_node_destination_vector.end(), phantom_node_vector.begin(), phantom_node_vector.end());
GetCoordinates(route_parameters.sources, phantom_node_source_vector);
GetCoordinates(route_parameters.destinations, phantom_node_destination_vector);